quiz.rb

```diff
@@ -2,19 +2,60 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
+class Dog
+
+  attr_reader :name
+  attr_reader :breed
+  attr_reader :age
+
+  def initialize(name,breed,age)
+    @name = name
+    @breed = breed
+    @age = age
+  end
+
+end
+
 # Question 2
 # Create a new Dog object.
 
+parker = Dog.new("Parker","Eskimo Lab","4")
+
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
+class Person
+
+  attr_writer :name
+  attr_reader :age
+  attr_writer :location
+
+  def initialize(name,age,location)
+    @name = name
+    @age = age
+    @location = location
+  end
+
+end
+
 # Question 4
 # Create a new Person object using parameters "Dave", 32, and "Ohio".
 # Then, set that Person's location to "Somerville"
 
+dave = Person.new("Dave","32","Ohio")
+dave.set_location("Somerville")
+
 # Question 5
 # Create another class called Developer that inherits from Person.
 # Give it a new private method called 'write_code', which prints out "TYPE TYPE TYPE".
 # Also give it a new public method called 'hire_for_job', which calls 'write_code'
+
+class Developer < Person
+
+  def write_code
```

Comment:
You've got the write_code method! But to make it private, we'll need to use the keyword `private`; any methods we define after private will be accessible only from within the object. Methods defined before the `private` will all be public.
So, in order for `hire_for_job` to be public and `write_code` to be private, we could write:
```ruby
  def hire_for_job
    ...
  end
  private
  def write_code
    ...
  end
```